BraveRewardsUI/Wallet/Notifications/NotificationHandler.swift

```diff
@@ -0,0 +1,192 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import BraveRewards
+
+protocol NotificationHandlerDelegate: AnyObject {
+  func show(notificationView: WalletNotificationView)
+  func handleAction(notification: RewardsNotification)
+  func hideNotifications()
+}
+
+class NotificationHandler {
+
+  private var ledger: BraveLedger
+  weak var delegate: NotificationHandlerDelegate?
+  
+  init(ledger: BraveLedger) {
+    self.ledger = ledger
+  }
+  
+  deinit {
+    stop()
+  }
+  
+  weak var currrentNotification: RewardsNotification?
+  
+  func start() {
+    // Add observer
+    stop()
+    NotificationCenter.default.addObserver(self, selector: #selector(notificationAdded(_:)), name: .init(rawValue: "BATBraveLedgerNotificationAdded"), object: nil)
+    loadNotification()
+  }
+  
+  private func stop() {
+    NotificationCenter.default.removeObserver(self, name: .init(rawValue: "BATBraveLedgerNotificationAdded"), object: nil)
+  }
+  
+  @objc private func notificationAdded(_ notification: Notification) {
+    // TODO: Filter notification?
+    if currrentNotification == nil {
+      loadNotification()
+    }
+  }
+  
+  private func loadNotification() {
+    guard delegate != nil else {
+      return
```

Comment:
It's not on the class to worry about whether it has a delegate or not. shouldn't return just because its not set. If its vital to be set to have it work properly, it should be required in `init`

Reply:
Removed, delegate must be set in `start()` for the handler to return callbacks.